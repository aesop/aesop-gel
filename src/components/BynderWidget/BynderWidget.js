import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import useScript from '~/customHooks/useScript';
import styles from './BynderWidget.module.css';

const BynderWidget = ({ className, heading, id }) => {
  useScript(
    'https://d8ejoa1fys2rk.cloudfront.net/bynder-embed/latest/bynder-embed.js',
    'bynder-widgets-js',
    {
      accountUrl: 'assetportal.aesop.com',
      language: 'en',
    },
  );

  const classSet = cx(styles.base, className, styles[parent]);

  return (
    <article className={classSet} id={id}>
      <h3 className={styles.heading}>{heading}</h3>
      <div
        data-bynder-widget="media-list"
        data-collection-id={id}
        data-rows="3"
      />
    </article>
  );
};

BynderWidget.propTypes = {
  className: PropTypes.string,
  heading: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

BynderWidget.defaultProps = {
  className: undefined,
  heading: undefined,
  id: undefined,
};

export default BynderWidget;
