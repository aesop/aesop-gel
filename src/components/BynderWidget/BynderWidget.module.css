@import 'styles/variables/index';
@import url('https://d8ejoa1fys2rk.cloudfront.net/bynder-embed/2.13/css/bynder-widgets.min.css');

.heading {
  position: absolute;
  z-index: 9;
  margin-top: 16px;
  margin-left: 18px;
  font-size: rem(15px);
  font-weight: bold;
}

.base {
  :global {
    .bynder-widget-media-list {
      border: 1px solid var(--color-grey-65) !important;
      margin-bottom: 50px;
      border-radius: 0 !important;
    }

    .bynder-widget-media-list .bynder-widget-header,
    .bynder-widget-media-list .bynder-widget-content {
      border: none !important;
      background-color: var(--color-white) !important;
      box-shadow: none !important;
    }

    a.bynder-widget-btn {
      border: 1px solid var(--color-secondary) !important;
      background-color: var(--color-secondary) !important;
      border-radius: 0 !important;
      box-shadow: 0 0 var(--color-secondary) !important;
    }

    a.bynder-widget-btn:hover {
      background-color: var(--color-secondary) !important;
    }

    div.media-item a.bynder-widget-media-link,
    ul.bynder-widget-media-items li a.bynder-widget-media-link {
      width: 80% !important;
      border: 1px solid var(--color-grey-65) !important;
      background-color: var(--color-white) !important;
      box-shadow: 0 0 transparent !important;
    }

    ul.bynder-widget-media-items li a.bynder-widget-media-link {
      border-radius: 0 !important;
    }

    div.media-item:hover a.bynder-widget-media-link,
    ul.bynder-widget-media-items li:hover a.bynder-widget-media-link {
      padding: 10px !important;
      border: 2px solid var(--color-secondary) !important;
      box-shadow: 0 0 transparent !important;
    }

    .bynder-widget-media-list .bynder-widget-header a.bynder-widget-link {
      visibility: hidden !important;
    }

    div.media-item a p,
    ul.bynder-widget-media-items li a p {
      width: 100% !important;
    }

    ul.bynder-widget-media-items li a .bynder-widget-mediaholder {
      width: 100% !important;
      background-color: transparent !important;
    }

    .bynder-widget-header a.bynder-widget-btn {
      right: 18px !important;
    }

    ul.bynder-widget-media-items li a.bynder-widget-btn {
      top: 0 !important;
      right: 0 !important;
      bottom: 0;
      left: 0;
      width: 70% !important;
      margin: auto !important;
    }

    ul.bynder-widget-media-items li a.bynder-widget-btn:hover {
      width: 70% !important;
    }

    .bynder-widget-media-list .bynder-widget-content {
      height: auto !important;
    }
  }
}
