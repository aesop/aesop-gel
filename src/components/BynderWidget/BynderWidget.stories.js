import React from 'react';
import { storiesOf } from '@storybook/react';
import '~.storybook/storybook.module.css';
import BynderWidget from './BynderWidget';

storiesOf('Components/BynderWidget', module).add('Base component', () => (
  <BynderWidget
    heading="Media Releases"
    id="6E8E63F9-7A54-442B-861E291124E19D94"
  />
));
