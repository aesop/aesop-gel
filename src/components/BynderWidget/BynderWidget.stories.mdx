import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import BynderWidget from './BynderWidget';
import BynderWidgetFixture from './BynderWidget.fixture';

<Meta title="Components/BynderWidget" component={BynderWidget} />

# BynderWidget

<Preview>
  <Story name="Base component">
    <BynderWidget
      heading="Media Releases"
      id="6E8E63F9-7A54-442B-861E291124E19D94"
    />
  </Story>
</Preview>

## API

<Props of={BynderWidget} />
