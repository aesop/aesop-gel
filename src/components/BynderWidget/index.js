import BynderWidget from './BynderWidget';

export default BynderWidget;
