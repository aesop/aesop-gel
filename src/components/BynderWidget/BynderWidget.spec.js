import React from 'react';
import { render } from '@testing-library/react';
import { BynderWidget } from './BynderWidget';

describe('<BynderWidget />', () => {
  it('should be defined', () => {
    expect(BynderWidget).toBeDefined();
  });

  it('should render base component correctly', () => {
    const { container } = render(
      <BynderWidget
        heading="Media Releases"
        id="6E8E63F9-7A54-442B-861E291124E19D94"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders null if there is an error', () => {
    const { container } = render(
      <BynderWidget heading="Media Releases" id="" />,
    );

    expect(container).toMatchSnapshot();
  });
});
