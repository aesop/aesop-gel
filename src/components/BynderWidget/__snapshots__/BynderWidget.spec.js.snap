// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BynderWidget /> renders base component correctly 1`] = `
<article
  className="base dark"
  id="6E8E63F9-7A54-442B-861E291124E19D94"
>
  <h3
    className="heading"
  >
    Media Releases
  </h3>
  <span
    className="base isLoading medium dark"
  >
    <span
      className="dot"
    />
    <span
      className="dot"
    />
    <span
      className="dot"
    />
  </span>
  <div
    data-bynder-widget="media-list"
    data-collection-id="6E8E63F9-7A54-442B-861E291124E19D94"
    data-rows="3"
  />
</article>
`;
